Loop/Extensions/CarbStore+SimulatedCoreData.swift

```diff
@@ -117,15 +117,13 @@ fileprivate extension StoredCarbEntry {
                                foodType: "Simulated",
                                absorptionTime: absorptionTime,
                                createdByCurrentApp: true,
-                               externalID: UUID().uuidString,
-                               isUploaded: false)
+                               externalID: UUID().uuidString)
```

Comment:
Just want to confirm that keeping this `externalID` was intended.

Reply:
Yes, `externalID` is needed for the DIY user that transitions from pre-RDS to post-RDS DIY Loop. Pre-RDS, the `externalID` field is used by Nightscout to "remember" the Nightscout entry ID (used for later updates/deletes). The post-RDS Nightscout code will use this field (if it exists) just for updates/deletes of previously existing data, but it will no longer store an externalID into that field (and, in fact, it cannot) for new data.

In short, it exists for backwards compatibility for DIY Loop users for migration purposes.